web/api/v1/api.go

```diff
@@ -310,10 +310,12 @@ func (api *API) query(r *http.Request) apiFuncResult {
 func (api *API) queryRange(r *http.Request) apiFuncResult {
 	start, err := parseTime(r.FormValue("start"))
 	if err != nil {
+		err = errors.New("missing parameter 'start'")
```

Comment:
Currently this does `fmt.Errorf("cannot parse %q to a valid timestamp", s)` in `parseTime()`, so completely overwriting that error here would lose the actual timestamp value that couldn't be parsed. Also, the parameter is not technically missing, but invalid (can't be parsed). How about just:

`err = fmt.Errorf("invalid parameter 'start': %s", err)`

Similar for the instance below.